api/src/main/java/ai/djl/util/Utils.java

```diff
@@ -349,4 +362,118 @@ public static Path getCacheDir() {
         }
         return Paths.get(cacheDir);
     }
+
+    /**
+     * scan classes files from a path to see if there is a matching implementation for a class.
+     *
+     * <p>For .class file, this function expects them in classes/your/package/ClassName.class
+     *
+     * @param path the path to scan from
+     * @param className the name of the classes, pass null if name is unknown
+     * @param <T> the Template T for the output Class
+     * @return the Class implementation
+     */
+    public static <T> T findImplementation(Path path, String className) {
+        try {
+            Path classesDir = path.resolve("classes");
+            // we only consider .class files and skip .java files
+            List<Path> jarFiles =
+                    Files.list(path)
+                            .filter(p -> p.toString().endsWith(".jar"))
```

Comment:
it it possible we have some other type of file as well ?

Reply:
We only scan on the .class files. These mostly only live in Java jars.